src/main/java/sentinel/utils/Parser.java

```diff
@@ -0,0 +1,154 @@
+package sentinel.utils;
+
+import sentinel.api.GeminiApi;
+import sentinel.exception.DeadlineException;
+import sentinel.exception.EventException;
+import sentinel.exception.SentinelException;
+import sentinel.task.Deadline;
+import sentinel.task.Event;
+import sentinel.task.Task;
+import sentinel.task.ToDo;
+import sentinel.ui.Ui;
+import sentinel.Sentinel;
+import java.time.LocalDateTime;
+import java.util.Arrays;
+
+/**
+ * The Parser class is responsible for parsing user input and converting it into commands or tasks.
+ * It includes methods for parsing command types, task names, deadlines, events, and indices.
+ */
+public class Parser {
+
+    /**
+     * Parses the user input to determine the command type.
+     *
+     * @param input The user's input string.
+     * @return The command type determined from the input.
+     */
+    public static Sentinel.CommandType parseForCommand(String input) {
+        String[] arr = input.split(" ");
+        return Sentinel.CommandType.valueOf(arr[0].toLowerCase());
+    }
+
+    /**
+     * Parses the user input to create a Task object based on the command type.
+     *
+     * @param commandType The type of command (e.g., todo, deadline, event).
+     * @param input The user's input string.
+     * @param ui The user interface object for displaying messages.
+     * @return A Task object if the input is valid; null otherwise.
+     * @throws SentinelException If the input is invalid or the task cannot be created.
+     */
+    public static Task parseTask(Sentinel.CommandType commandType, String input, Ui ui) throws SentinelException {
+        String taskName = parseTaskName(commandType, input, ui);
+        if (taskName.isEmpty()) {
+            ui.showEmptyTaskNameError(commandType);
+            return null;
+        }
+
+        switch (commandType) {
```

Comment:
Case indentation does not follow course guidelines here 😢

Reply:
Thank you for highlighting this! I'd changed it on a more recent commit after someone brought it up on the forums but haven't opened a PR yet. Thank you for reminding me!